File: python_ruby/LinkedList.rb
class Node
  attr_accessor :data, :next_
  def initialize(data,next_ = nil)
    @data = data
    @next_ = next_
  end
  def to_s
    return @data.to_s
  end
end

#basic example
head = Node.new(0)
head.next_ = Node.new(1)
puts head.next_

head = Node.new(0)
cur = head
for i in 0..10000
  new_node = Node.new(i)
  cur.next_ = new_node
  cur = cur.next_
end

cur = head
while cur
  puts cur
  cur = cur.next_
end

File: python_ruby/LinkedList.py
class Node:
    def __init__(self,data,next=None):
        self.data = data
        self.next = next
    def __str__(self):
        return repr(self.data)

#basic example
head = Node(0)
head.next = Node(1)
print head.next

#real linked list
head = Node(0)
cur = head
for i in xrange(10000):
    new_node = Node(i)
    cur.next = new_node
    cur = cur.next

cur = head
while cur:
    print cur
    cur = cur.next
